resteasy-client-reactor-netty/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ReactorNettyClientHttpEngine.java

```diff
@@ -126,8 +152,11 @@ public ReactorNettyClientHttpEngine(final HttpClient httpClient,
                                                         request.getClientConfiguration(),
                                                         response,
                                                         null,
-                                                        extractor)))))
-                .toFuture();
+                                                        extractor)))));
+
+        return timeout
+                .map(timeout -> responseMono.timeout(timeout))
```

Comment:
I think it would read better with `.map(duration -> responseMono.timeout(duration)))`.